server/src/lib/server.rs

```diff
@@ -128,6 +128,24 @@ fn propolis_to_api_state(
     }
 }
 
+// This is a somewhat hard-coded translation of a stable "PCI slot" to a BDF.
+//
+// For all the devices requested by Nexus (network interfaces, disks, etc),
+// we'd like to assign a stable PCI slot, such that re-allocating these
+// devices on a new instance of propolis produces the same guest-visible
+// BDFs.
+fn slot_to_bdf(slot: api::PciSlot) -> Result<pci::Bdf> {
+    match slot.0 {
+        // NOTE: This "5" is 100% arbitrary; it just shows we'd map to different
+        // slots as new devices are requested.
+        n if n < 5 => Ok(pci::Bdf::new(0, n, 0)),
```

Comment:
Slots 0 and 1 are taken by the hostbridge and chipset, respectively.  I might make sense for each class of devices (NICs, block volumes, etc) to each have its own index and corresponding mapping.  Perhaps this?:
0x8 - 0xf: NICs 0-7
0x10 - 0x17 - blockdevs 0.7

Reply:
Updated to use this slot mapping scheme.